src/pkg/k8s/pods.go

```diff
@@ -72,111 +79,125 @@ func (k *K8s) DeletePods(target PodLookup) error {
 }
 
 // CreatePod inserts the given pod into the cluster.
-func (k *K8s) CreatePod(pod *corev1.Pod) (*corev1.Pod, error) {
+func (k *K8s) CreatePod(ctx context.Context, pod *corev1.Pod) (*corev1.Pod, error) {
 	createOptions := metav1.CreateOptions{}
-	return k.Clientset.CoreV1().Pods(pod.Namespace).Create(context.TODO(), pod, createOptions)
+	return k.Clientset.CoreV1().Pods(pod.Namespace).Create(ctx, pod, createOptions)
 }
 
 // GetAllPods returns a list of pods from the cluster for all namespaces.
-func (k *K8s) GetAllPods() (*corev1.PodList, error) {
-	return k.GetPods(corev1.NamespaceAll, metav1.ListOptions{})
+func (k *K8s) GetAllPods(ctx context.Context) (*corev1.PodList, error) {
+	return k.GetPods(ctx, corev1.NamespaceAll, metav1.ListOptions{})
 }
 
 // GetPods returns a list of pods from the cluster by namespace.
-func (k *K8s) GetPods(namespace string, listOpts metav1.ListOptions) (*corev1.PodList, error) {
-	return k.Clientset.CoreV1().Pods(namespace).List(context.TODO(), listOpts)
+func (k *K8s) GetPods(ctx context.Context, namespace string, listOpts metav1.ListOptions) (*corev1.PodList, error) {
+	return k.Clientset.CoreV1().Pods(namespace).List(ctx, listOpts)
 }
 
 // WaitForPodsAndContainers attempts to find pods matching the given selector and optional inclusion filter
 // It will wait up to 90 seconds for the pods to be found and will return a list of matching pod names
 // If the timeout is reached, an empty list will be returned.
-func (k *K8s) WaitForPodsAndContainers(target PodLookup, include PodFilter) []corev1.Pod {
-	for count := 0; count < waitLimit; count++ {
+func (k *K8s) WaitForPodsAndContainers(ctx context.Context, target PodLookup, include PodFilter) []corev1.Pod {
+	waitCtx, cancel := context.WithTimeout(ctx, 90*time.Second)
+	defer cancel()
 
-		pods, err := k.Clientset.CoreV1().Pods(target.Namespace).List(context.TODO(), metav1.ListOptions{
-			LabelSelector: target.Selector,
-		})
-		if err != nil {
-			k.Log("Unable to find matching pods: %w", err)
-			break
-		}
+	timer := time.NewTimer(0)
+	defer timer.Stop()
 
-		k.Log("Found %d pods for target %#v", len(pods.Items), target)
-
-		var readyPods = []corev1.Pod{}
+	for {
+		select {
+		case <-waitCtx.Done():
+			k.Log("Pod lookup timeout or context cancelled: %w", ctx.Err())
```

Comment:
```suggestion
			k.Log("Pod lookup failed: %w", ctx.Err())
```
timeout or context cancelled should already be clear from the ctx.Err()